### PR TITLE
Fix check-kubernetes to wait for pods properly

### PR DESCRIPTION
**Pull request by stefwalter**
Otherwise this causes races during testing

### CODE REVIEW

File: pkg/kubernetes/client.js [hunk 1]
```diff
@@ -415,7 +415,7 @@ define([
 
                     update_error_message(ex, response);
                     var msg = "watching kubernetes " + type + " failed: " + ex.message;
-                    if (ex.problem !== "disconnected")
+                    if (ex.problem !== "disconnected" && ex.problem !== "not-found")
```
Comment (petervo) [inline pkg/kubernetes/client.js]: This doesn't stop the message for me, ex.problem can be null for 404s. Below we are checking ex.status to determine if it's a 404 that might work here too.

Comment (stefwalter) [inline pkg/kubernetes/client.js]: Right. Duh. Sorry

**Comment by stefwalter** [comment]
The Travis failure is #2845, rebased on that.

**Comment by stefwalter** [comment]
F22 failure related to #2759

**Pull request closed without merging**